components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/util/IdentityCommonUtils.java

```diff
@@ -0,0 +1,285 @@
+                    digestAlgorithm);
+        }
+        //generating hash value
+        md.update(value.getBytes(StandardCharsets.UTF_8));
+        byte[] digest = md.digest();
+        int leftHalfBytes = digest.length / 2;
+        byte[] leftmost = new byte[leftHalfBytes];
+        System.arraycopy(digest, 0, leftmost, 0, leftHalfBytes);
+
+        return Base64.getUrlEncoder().withoutPadding().encodeToString(leftmost)
+                .replace("\n", "").replace("\r", "");
+    }
+
+    /**
+     * Get the configured certificate header name.
+     *
+     * @return value of the cert header name configuration
+     */
+    public static String getMTLSAuthHeader() {
+
+        return Optional.ofNullable(IdentityUtil.getProperty(IdentityCommonConstants.MTLS_AUTH_HEADER))
+                .orElse("CONFIG_NOT_FOUND");
+    }
+
+    /**
+     * Parse the certificate content.
+     *
+     * @param content the content to be passed
+     * @return the parsed certificate
+     * @throws ConsentManagementException  if an error occurs while parsing the certificate
+     */
+    public static X509Certificate parseCertificate(String content) throws ConsentManagementException {
+
+        try {
+            if (StringUtils.isNotBlank(content)) {
+                // removing illegal base64 characters before decoding
+                content = removeIllegalBase64Characters(content);
+                byte[] bytes = Base64.getDecoder().decode(content);
+
+                return (java.security.cert.X509Certificate) CertificateFactory.getInstance(IdentityCommonConstants.X509)
+                        .generateCertificate(new ByteArrayInputStream(bytes));
+            }
+            log.error("Certificate passed through the request is empty");
+            return null;
+        } catch (CertificateException | IllegalArgumentException e) {
+            throw new ConsentManagementException("Certificate passed through the request not valid", e);
+        }
+    }
+
+    /**
+     * Remove illegal base64 characters from input string.
+     *
+     * @param value certificate as a string
+     * @return certificate without illegal base64 characters
+     */
+    private static String removeIllegalBase64Characters(String value) {
+        if (value.contains(IdentityCommonConstants.BEGIN_CERT)
+                && value.contains(IdentityCommonConstants.END_CERT)) {
+
+            // extracting certificate content
+            value = value.substring(value.indexOf(IdentityCommonConstants.BEGIN_CERT)
+                            + IdentityCommonConstants.BEGIN_CERT.length(),
+                    value.indexOf(IdentityCommonConstants.END_CERT));
+        }
+        // remove spaces, \r, \\r, \n, \\n, ], [ characters from certificate string
+        return value.replaceAll("\\\\r|\\\\n|\\r|\\n|\\[|]| ", StringUtils.EMPTY);
+    }
+
+    /**
+     * Utility method get the application property from SP Meta Data.
```

Comment:
Need to update the comment to match the logic.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/62253df1607d32bdebcd905b7140f43984f3c36a